src/language-js/printer-estree.js

```diff
@@ -5285,22 +5285,79 @@ function isObjectType(n) {
 // eg; `describe("some string", (done) => {})`
 function isTestCall(n) {
   const unitTestRe = /^(skip|(f|x)?(it|describe|test))$/;
+
+  /*
+  * The below check makes sure that it('very long', async(()=>{//code}))
+  * async(()) doesn't break
+  */
+
+  if (n.arguments.length === 1) {
+    if (n.callee.type === "Identifier" && n.callee.name === "async") {
+      return isFunctionOrArrowExpression(n.arguments[0].type);
+    }
+
+    if (isUnitTestSetUp(n)) {
+      return (
+        isFunctionOrArrowExpression(n.arguments[0].type) ||
+        isIdentiferAsync(n.arguments[0])
+      );
+    }
+    return false;
+  } else if (n.arguments.length === 2) {
+    if (
+      (n.callee.type === "Identifier" && unitTestRe.test(n.callee.name)) ||
+      isSkipOrOnlyBlock(n)
+    ) {
+      if (n.arguments.length === 2) {
+        if (
+          isTemplateLiteral(n.arguments[0]) ||
+          isStringLiteral(n.arguments[0])
```

Comment:
Why nested `if` but for the only possible branch?

cc @j-f1: I'm not sure if this is what you suggested in https://github.com/prettier/prettier/pull/4010#discussion_r169424425.

Reply:
I just meant splitting up the multiple sections joined by `&&` so that people could read them.